### DISCUSSION TITLE
Remove base64 images from __NEXT_DATA__

### DISCUSSION THREAD

**Discussion by prKassad** [vercel/next.js#13693] [Help]
I'am using SSR and getting base64 images from API in getInitialProps. However, base64 images are also duplicated in props` __NEXT_DATA__`. This increases page size and has a bad effect on SEO metrics. Is it possible to remove duplicate images data from `__NEXT_DATA__`?

**Comment by jamesmosier** [comment]
Next populates the data when server side rendering from `__NEXT_DATA__`. Without that, your page would not contain any content when server rendered 😄

**Reply by prKassad** [reply]
Images are already in HTML which rendered nextjs. Really without duplicate img-data Hydration will not pass? Or there is no way to modify `__NEXT_DATA__` after HTML rendering by server? 🙄

**Comment by prKassad** [comment]
Hmm... This code works... And hydrate successful
```
export default class MyDocument extends Document<any> {
  constructor(props) {
    super(props);
    const nextData = props.__NEXT_DATA__;
    // change images on next data
    nextData.props.pageProps.base64image = "data:image/jpeg;base64,";
  }
...
```
Thus we have image only in rendered HTML.

**Reply by prKassad** [reply]
UPD: there are hydration errors 😀
